### PR TITLE
DFPL-1182 Address fields not getting reset when 'Reason address not Known' Radio is selected

### PR DESCRIPTION
**Pull request by KatKovacs1**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DFPL-1182

### Change description ###

On enterOthers, set address field of Other to null when address not known and set Reason address not Known to null when address is known.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OthersControllerAboutToSubmitTest.java [hunk 1]
```diff
@@ -88,18 +88,36 @@ void shouldNotSaveOtherWhenAllOtherElementsAreNullOrEmpty() {
...
+                .firstOther(other())
+                .additionalOthers(additionalOthersWithoutAddress())
+                .build()))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToSubmitEvent(caseDetails);
+        CaseData caseData = mapper.convertValue(response.getData(), CaseData.class);
+
+        assertThat(unwrapElements(caseData.getConfidentialOthers())).contains(confidentialOther());
+        assertThat(unwrapElements(caseData.getConfidentialOthers())).contains(confidentialOtherWithoutAddress());
```
Comment (mapingo) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OthersControllerAboutToSubmitTest.java:104]: The .contains(confidentialOtherWithoutAddress()) can be chained to the line above

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/Other.java [hunk 2]
```diff
@@ -158,4 +158,19 @@ public boolean isDeceasedOrNFA() {
         return AddressNotKnowReason.DECEASED.getType().equals(addressNotKnowReason)
             || AddressNotKnowReason.NO_FIXED_ABODE.getType().equals(addressNotKnowReason);
     }
+
+    public Other removeAddress() {
+        Other other =  this.toBuilder()
```
Comment (tomxelliott) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/Other.java:163]: You can just return the builder
`return this.toBuilder()
.address(null)
.build();
`

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/Other.java [hunk 3]
```diff
@@ -158,4 +158,19 @@ public boolean isDeceasedOrNFA() {
...
+            .address(null)
+            .build();
+        return other;
+    }
+
+    public Other removeAddressNotKnowReason() {
+        Other other =  this.toBuilder()
```
Comment (tomxelliott) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/Other.java:170]: Same here as above, just return the builder call. No need to create a variable then return the variable

**Review by tomxelliott — CHANGES_REQUESTED** [review]
Please action the 2 requests then will approve

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/Other.java (hunk 2 above)
Comment (KatKovacs1) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/Other.java]: changed

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OthersControllerAboutToSubmitTest.java (hunk 1 above)
Comment (KatKovacs1) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OthersControllerAboutToSubmitTest.java:104]: changed

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/Other.java (hunk 3 above)
Comment (KatKovacs1) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/Other.java]: changed

**Review by mapingo — APPROVED** [review]

**Pull request merged**